engine/executor/dm/api.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dm
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pingcap/errors"
+
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/runtime"
+	dmpkg "github.com/pingcap/tiflow/engine/pkg/dm"
+)
+
+// QueryStatus implements the api of query status request.
+func (t *baseTask) QueryStatus(ctx context.Context, req dmpkg.QueryStatusRequest) (dmpkg.QueryStatusResponse, error) {
+	// get status from unit
+	status := t.unitHolder.Status(ctx)
+	// copy status via json
+	statusBytes, err := json.Marshal(status)
+	if err != nil {
+		return dmpkg.QueryStatusResponse{ErrorMsg: err.Error()}, nil
+	}
+	taskStatus := runtime.NewTaskStatus(t.workerType, t.taskID, t.stage)
+	if err = json.Unmarshal(statusBytes, taskStatus); err != nil {
+		return dmpkg.QueryStatusResponse{ErrorMsg: err.Error()}, nil
+	}
+	return dmpkg.QueryStatusResponse{TaskStatus: taskStatus}, nil
+}
+
+// StopWorker implements the api of stop worker message which kill itself.
+func (t *baseTask) StopWorker(ctx context.Context, msg dmpkg.StopWorkerMessage) error {
```

Comment:
or here you want to express the relationship of "task has worker so task can stop its worker"?

Reply:
this is align with createWorker 😂